### PR TITLE
[Auton-GUI] Added functionality for offline map viewing

### PR DESCRIPTION
**Pull request by jnnanni**
image
The checkbox in the bottom left of the map will toggle whether the map uses the openstreetmap webapi or the static/map directory to use local tiles. 
Google doc explaining setup:
https://docs.google.com/document/d/1gc2yv4KhZgtrdS6AzUc37DAglwTzC9N3XXtQOB86pQw/edit

### CODE REVIEW

**Comment by cgiger00** [comment]
is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover

**Comment by qhdwight** [comment]
> is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover

Sometimes I think it makes sense to have a MRover S3 bucket. We could store stuff like this, pointclouds, built packages, etc.

**Comment by jnnanni** [comment]
> is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover

My solution to this would be to make a batch file in the base-station/gui/static folder that will download the map folder to the same directory. Relative directories with javascript are kind of a hassle when accessing files outside of the js root directory so .mrover might not be the best spot for the tilemap. Does this sound like a good solution?

**Comment by cgiger00** [comment]
> > is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover
> 
> Sometimes when things like this pops up I think it makes sense to have a MRover Amazon S3 bucket. We could store stuff like this, pointclouds, built packages, etc.

how difficult would that be?

**Comment by cgiger00** [comment]
> > is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover
> 
> My solution to this would be to make a batch file in the base-station/gui/static folder that will download the map folder to the same directory. Relative directories with javascript are kind of a hassle when accessing files outside of the js root directory so .mrover might not be the best spot for the tilemap. Does this sound like a good solution?

i guess this is a workable temporary solution, do the thinkpads have enough space for the file download? can you then add something to a .gitignore so they don't get accidentally pushed to github?

**Comment by jnnanni** [comment]
> > > is there a way to do this without permanently adding 3000 files to the git repo? like something that downloads a portion of them to .mrover
> > 
> > 
> > My solution to this would be to make a batch file in the base-station/gui/static folder that will download the map folder to the same directory. Relative directories with javascript are kind of a hassle when accessing files outside of the js root directory so .mrover might not be the best spot for the tilemap. Does this sound like a good solution?
> 
> i guess this is a workable temporary solution, do the thinkpads have enough space for the file download? can you then add something to a .gitignore so they don't get accidentally pushed to github?

The file size of the current files is only ~25 MB, if we get just the map of Ann Arbor and Utah it will likely be less than 100MB, the thinkpads hopefully shouldn't have any issues with that.

**Comment by cgiger00** [comment]
stalled on getting s3 account

**Comment by CameronTressler** [comment]
@jnnanni can you resolve the conflict for this pr in the meantime

**Comment by jnnanni** [comment]
> 

All Set

**Comment by cgiger00** [comment]
should i turn this into a draft?

**Comment by CameronTressler** [comment]
For now we can, I know @jnnanni was very close to pushing another commit though

**Comment by jnnanni** [comment]
This is now all ready and tested.

**Comment by jnnanni** [comment]
Just tested with the new ssh keys, works

File: base_station/download_map/src/__main__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import boto3
```
Comment (cgiger00) [inline base_station/download_map/src/__main__.py:1]: you added all of these to requirements.txt?

File: pip_deps/requirements.txt [hunk 2]
```diff
@@ -5,3 +5,4 @@ pygobject==3.28.2
 pyusb==1.0.2
 Adafruit_BBIO==1.1.1
 smbus-cffi==0.5.1
+boto3==1.17.112
```
Comment (cgiger00) [inline pip_deps/requirements.txt:8]: all good without adding zipfile?

**Review by cgiger00 — APPROVED** [review]
lgtm assuming the deps are all good

File: pip_deps/requirements.txt (hunk 2 above)
Comment (jnnanni) [reply pip_deps/requirements.txt:8]: Yes, zipfile is a standard python package

**Comment by CameronTressler** [comment]
@jnnanni can you resolve the conflicts on this when you get a chance

@cgiger00 do you think we should PR this into travis-free now? Or just stick with auton-integration

**Comment by jnnanni** [comment]
> @jnnanni can you resolve the conflicts on this when you get a chance
Taken care of

**Comment by cgiger00** [comment]
lets do auton integration since travis free was merged into main

**Pull request merged**